pgjdbc/src/main/java/com/yugabyte/ysql/LoadBalanceService.java

```diff
@@ -99,6 +95,28 @@ private static synchronized boolean refresh(Connection conn, long refreshInterva
       String region = rs.getString("region");
       String zone = rs.getString("zone");
       String nodeType = rs.getString("node_type");
+      try{
+        uuid = rs.getString("universe_uuid");
+      } catch (Exception e){
+        LOGGER.warning("Connections being made to an older version of yugabytedb with universe uuid not present in yb_servers() function. Multi cluster connections are not supported in case of two such clusters");
+        uuid = "default";
+      }
+
+      if (lb.getUuid() == null){
+        lb.setUuid(uuid);
+      }
+      if (cluster == null){
+        if (uuidToClusterInfoMap.containsKey(uuid)){
+          cluster = uuidToClusterInfoMap.get(uuid);
+          conn.close();
+        } else {
+          cluster = new ClusterInfo();
+          cluster.setControlConnection(conn);
+        }
+      }
```

Comment:
Please see the comment in `checkAndRefresh()`.
(part 1):
```suggestion
      if (cluster == null){
        if (uuidToClusterInfoMap.containsKey(uuid)){
          cluster = uuidToClusterInfoMap.get(uuid);
        } else {
          cluster = new ClusterInfo();
          cluster.setControlConnection(conn);
        }
      }
```

(part 2) immediately after while loop:
```
    if (lb.getUuid() == null) {
      if (uuidToClusterInfoMap.containsKey(uuid)) {
        conn.close();
      }
      lb.setUuid(uuid);
    }
```